static/src/javascripts/projects/facia/modules/ui/container-show-more.js

```diff
@@ -47,32 +60,105 @@ define([
         var prefs = userPrefs.get(prefName, {
             type: 'session'
         });
-        return (prefs && prefs[containerId]) ? 'displayed' : 'hidden';
+        return (prefs && prefs[containerId]) ? STATE_DISPLAYED : STATE_HIDDEN;
     }
 
-    function showMore($container, button) {
+    function showMore(button) {
         fastdom.write(function () {
             /**
              * Do not remove: it should retain context for the click stream module, which recurses upwards through
              * DOM nodes.
              */
-            $container.toggleClass(className, button.state === 'displayed');
-            button.state = (button.state === 'hidden') ? 'displayed' : 'hidden';
-            setButtonState(button, button.state);
+            setButtonState(button, (button.state === STATE_HIDDEN) ? STATE_DISPLAYED : STATE_HIDDEN);
             updatePref(button.id, button.state);
         });
     }
 
-    function renderToDom($container, button) {
+    function renderToDom(button) {
         fastdom.write(function () {
-            $container.addClass(className)
+            button.$container.addClass(className)
                 .removeClass('js-container--fc-show-more')
-                .toggleClass(className, button.state === 'hidden');
+                .toggleClass(className, button.state === STATE_HIDDEN);
```

Comment:
Also we can test by using the perf tool Patrick sent around - it shows you when reads & writes occur.